### PR TITLE
vim-patch:2f0936cb9a2e

### PR DESCRIPTION
**Pull request by clason**
Update runtime files
https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b

### CODE REVIEW

**Review by seandewar — COMMENTED** [review]
a lot of these are error number shuffles, (ah, just realized you skipped most :P) but we're missing:

- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-d097d8ca80d44024b9712876f6d4e21d2b6829d7be2886916c96ff0d2050c719R227 (this didn't auto apply because we also have E856 which is nvim only)
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-b3aeef96275167ddcc1e5fee6d51b8f2018f9359f921d86ad6e7035ca80f20e9R4950 and the other one below it
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-b3aeef96275167ddcc1e5fee6d51b8f2018f9359f921d86ad6e7035ca80f20e9R1349 (and E276)
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-b3aeef96275167ddcc1e5fee6d51b8f2018f9359f921d86ad6e7035ca80f20e9R175 (and E907)
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-55e70d883ea4295dff867a5ff35dca80d42926102379a4be39aa7d8b29d779c1R92
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-e10fa5167b70184f67ba6b6e3c015a16e570e5d67aa1d45fd97f938c50a06d87R247 (and below)
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-f17e0dd888d6e8cd63e14b598ed3a141e59870e04a35701b66c93ddb175f2a57R1102

although we're missing this patch, it might be easier to include this anyway:
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-b3aeef96275167ddcc1e5fee6d51b8f2018f9359f921d86ad6e7035ca80f20e9R2577 (need 8.2.0908)

also we still have this [using xim section](https://github.com/neovim/neovim/blob/f6a9f0bfcaeb256bac1b1c8273e6c40750664967/runtime/doc/mbyte.txt#L492) that was changed in https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-b5ce8ea08d78c0baef6b22f54deb0c81f940d56c6d8cf651b29ce950e9367568R865-R866 but i think it's N/A and should be removed? (probably not in this PR, i think there's other stuff in mbyte.txt that probably needs looking at anyway)

**Comment by jamessan** [comment]
> (this didn't auto apply because we also have E856 which is nvim only)

Hmm, that shouldn't be nvim only unless Vim removed it at some point and we didn't.  All of our error codes should be > 5000.

**Comment by seandewar** [comment]
> unless Vim removed it at some point and we didn't

I don't think so. Seems to be introduced a while ago in https://github.com/neovim/neovim/commit/73da522d73c5d686c122ea4482d03e576a6a242c (https://github.com/neovim/neovim/pull/5996) as an alternative to E854.

**Comment by clason** [comment]
Shuffled the error codes now, too :)

File: runtime/doc/eval.txt [hunk 1]
```diff
@@ -1195,6 +1195,7 @@ When expr8 is a |Funcref| type variable, invoke the function it refers to.
 expr8->name([args])	method call			*method* *->*
 expr8->{lambda}([args])
 
+							*E260*
```
Comment (seandewar) [inline runtime/doc/eval.txt:1198]: could also add E276: https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-b3aeef96275167ddcc1e5fee6d51b8f2018f9359f921d86ad6e7035ca80f20e9R1349, though it's not from this rt update (it's from an rt update from before method call syntax was ported so I had missed it)

File: runtime/doc/eval.txt [hunk 2]
```diff
@@ -6454,7 +6457,7 @@ join({list} [, {sep}])					*join()*
 		Can also be used as a |method|: >
 			mylist->join()
 
-json_decode({expr})					*json_decode()*
+json_decode({expr})				*json_decode()* *E491*
```
Comment (seandewar) [inline runtime/doc/eval.txt:6460]: nvim's json_decode implementation doesn't have this error

File: runtime/doc/eval.txt [hunk 3]
```diff
@@ -6894,7 +6895,7 @@ match({expr}, {pat} [, {start} [, {count}]])			*match()*
 			GetText()->match('word')
 			GetList()->match('word')
 <
-				*matchadd()* *E798* *E799* *E801* *E957*
+				*matchadd()* *E290* *E798* *E799* *E801* *E957*
```
Comment (seandewar) [inline runtime/doc/eval.txt:6898]: interestingly nvim uses E5031 here instead

seems that originally this error didn't have an error code, so it was given one in this commit: https://github.com/neovim/neovim/commit/fb07391ce46b8bff90ef7ef073b75919a307e64c that also added more info to the emsg. Vim noticed later on and assigned it to E290 :eyes:

**Review by seandewar — COMMENTED** [review]
missing:

- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-b3aeef96275167ddcc1e5fee6d51b8f2018f9359f921d86ad6e7035ca80f20e9R4958
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-b3aeef96275167ddcc1e5fee6d51b8f2018f9359f921d86ad6e7035ca80f20e9R175 (could also add E907)
- https://github.com/vim/vim/commit/2f0936cb9a2eb026acac03e6a8fd0b2a5d97508b#diff-f17e0dd888d6e8cd63e14b598ed3a141e59870e04a35701b66c93ddb175f2a57R1102 we also have this typo

**Review by seandewar — APPROVED** [review]
LGTM!

**Pull request merged**